### ISSUE TITLE
DefaultNodeExtension does not implement ServiceLoader provider constructor

### ISSUE THREAD

**Issue by CAFxX** [hazelcast/hazelcast#16596]
**Describe the bug**
While trying to compile hazelcast with graalvm native image, we get the [following error](https://github.com/oracle/graal/issues/1508#issuecomment-559941197):

```
Class com.hazelcast.instance.DefaultNodeExtension cannot be instantiated reflectively . It does not have a nullary constructor.
```

The graalvm devs have [narrowed it down to an issue with `DefaultNodeExtension`](https://github.com/oracle/graal/issues/1508#issuecomment-562259645):

> The underlying problem is that com.hazelcast.instance.DefaultNodeExtension *should* provide a nullary constructor because it is registered as a service provider for com.hazelcast.instance.impl.NodeExtension. According to the ServiceLoader documentation The only requirement enforced by this facility is that provider classes must have a zero-argument constructor so that they can be instantiated during loading.

**Expected behavior**

I expect compilation to succeed without resorting to disable `ServiceLoaderFeature`.

**To Reproduce**

Steps to reproduce the behavior: see Dockerfile provided in https://github.com/oracle/graal/issues/1508#issue-470182861

**Comment by mmedenjak** [comment]
Thank you for the report @CAFxX !

Seems simple to fix and we'll take a look. Maybe you'd like to provide the PR yourself? We definitely appreciate it :)

**Comment by jerrinot** [comment]
hi @CAFxX, first of all: Thank you for trying Hazelcast with GraalVM Native Image!
@pivovarit is working on [Hazecast - Quarkus integration](https://github.com/quarkusio/quarkus/pull/6758) so perhaps you want to get in touch with him. As I reckon a lot of issues he is dealing with are related to NativeImage. 

Hazelcast does not use the ServiceLoader mechanism to load extensions. 
GraalVM NativeImage builder is trying to be smart and it detects everything in `META-INF/services` as ServiceLoader. A quick hack to try could be to actually add the argument-less constructor. That would make GraalVM plugin happy. You would still have to register the other constructor for reflective invocation. The right solution would be to a different directory name. However this is problematic for compatibility reasons. 

I believe e.g. the Quarkus framework also use `-H:-UseServiceLoaderFeature`, probably for similar reasons.

**Comment by pivovarit** [comment]
Just like @jerrinot said above. We could hack it through, but it's not even 1% needed to get the server part up and running on native. This and hundreds of other small issues will be taken care of when we start working on enabling GraalVM compatibility. I added a _GraalVM_ label for future reference.

Closing for now.

Btw, the client part works smoothly with Quarkus already([link](https://github.com/hazelcast/quarkus-hazelcast-client))
